code/modules/research/server.dm

```diff
@@ -52,14 +71,22 @@
 	stat &= ~EMPED
 	refresh_working()
 
+/obj/machinery/rnd/server/proc/toggle_disable()
+	research_disabled = !research_disabled
+	refresh_working()
+
 /obj/machinery/rnd/server/proc/mine()
 	. = base_mining_income
 	var/penalty = max((get_env_temp() - temp_tolerance_high), 0) * temp_penalty_coefficient
+	current_temp = get_env_temp()
 	. = max(. - penalty, 0)
 
 /obj/machinery/rnd/server/proc/get_env_temp()
-	var/datum/gas_mixture/environment = loc.return_air()
-	return environment.temperature
+	//var/datum/gas_mixture/environment = loc.return_air()
```

Comment:
Why this change ?

Reply:
The old code runtimes.

As far as I can tell, many of these procs aren't even used anymore. I suspect that they are relics from before techweb. My guess is that between that and now, atmos code was updated and return_air() no longer returns temperature too.

I change it specifically because I use it to get the temperature for the console readout. In the near future, I want to make servers produce heat again, and slow down if it gets too hot, but I didn't want to hang this PR up on that.

Reply:
Another idea for the heat production would be the ability to overclock or under clock the server for point generation but at the cost of more heat